svm/src/account_loader.rs

```diff
@@ -229,48 +203,19 @@ fn load_transaction_accounts<CB: TransactionProcessingCallback>(
                         .get_account_shared_data(key)
-                        .map(|mut account| {
-                            if message.is_writable(i) {
-                                if !feature_set
-                                    .is_active(&feature_set::disable_rent_fees_collection::id())
-                                {
-                                    let rent_due = rent_collector
-                                        .collect_from_existing_account(key, &mut account)
-                                        .rent_amount;
-
-                                    (account.data().len(), account, rent_due)
-                                } else {
-                                    // When rent fee collection is disabled, we won't collect rent for any account. If there
-                                    // are any rent paying accounts, their `rent_epoch` won't change either. However, if the
-                                    // account itself is rent-exempted but its `rent_epoch` is not u64::MAX, we will set its
-                                    // `rent_epoch` to u64::MAX. In such case, the behavior stays the same as before.
-                                    if account.rent_epoch() != RENT_EXEMPT_RENT_EPOCH
-                                        && rent_collector.get_rent_due(
-                                            account.lamports(),
-                                            account.data().len(),
-                                            account.rent_epoch(),
-                                        ) == RentDue::Exempt
-                                    {
-                                        account.set_rent_epoch(RENT_EXEMPT_RENT_EPOCH);
-                                    }
-                                    (account.data().len(), account, 0)
-                                }
-                            } else {
-                                (account.data().len(), account, 0)
-                            }
-                        })
+                        .map(|account| (account.data().len(), account))
                         .unwrap_or_else(|| {
                             account_found = false;
                             let mut default_account = AccountSharedData::default();
                             // All new accounts must be rent-exempt (enforced in Bank::execute_loaded_transaction).
                             // Currently, rent collection sets rent_epoch to u64::MAX, but initializing the account
                             // with this field already set would allow us to skip rent collection for these accounts.
                             default_account.set_rent_epoch(RENT_EXEMPT_RENT_EPOCH);
-                            (default_account.data().len(), default_account, 0)
+                            (default_account.data().len(), default_account)
                         })
                 };
                 accumulate_and_check_loaded_account_data_size(
```

Comment:
think these checks would need to be pulled out as well. it seems possible an account's size if modified by tx1 which would cause tx2 to fail these checks.

Reply:
`accumulate_and_check_loaded_account_data_size` accumulates the size of data in all accounts loaded by a transaction. If it exceeds the data budget, the tx loading would fail.
But account size could be modified by a previous transaction in the batch

Reply:
Looks like this only got half picked up, I see the checks added in immediately before execution. But we cannot do them here; account size could also be reduced by a transaction, making a later transaction no longer violate budget. Doing it here can break consensus

Reply:
So, if I understand it correctly, account size checking should be removed from account loading function and moved to the execution (which I have already done).
The only change required here is to remove it from here i.e `load_transaction_accounts`, right?

Reply:
Right. Any checks on account state (lamports, data size) should be done immediately before execution of that specific tx, because any previous tx in batch (may) have changed the result of that check.